### DISCUSSION TITLE
error message for installation

### DISCUSSION THREAD

**Discussion by snugook1108** [idaholab/moose#19667] [Q&A Getting Started] [answered]
```
/home/1ox/anaconda3/envs/moose/bin/../lib/gcc/x86_64-conda-linux-gnu/9.3.0/../../../../x86_64-conda-linux-gnu/bin/ld: /home/1ox/project_mole_test/mole/src/.libs/main.x86_64-conda-linux-gnu.opt.o: in function `main':
main.C:(.text.startup.main+0x3b): undefined reference to `XXXTestApp::registerApps()'
collect2: error: ld returned 1 exit status
```
The undefined reference is there. I don't know how to solve this problem.

**Comment by GiudGiud** [comment, marked as answer]
This looks like remains from a previous compilation. Try to `make clobberall` then if that fails, commit/save all your work and use `git clean -xfd`. You may want to do this for every submodule with this: `git submodule foreach --recursive git clean -xfd`

**Reply by snugook1108** [reply]
I tried to use ```git clean -xfd``` and ```git submodule foreach --recursive git clean -xfd``` , and ```make clobberall``` but the same message, ```main.C:(.text.startup.main+0x3b): undefined reference to `XXXTestApp::registerApps()'
collect2: error: ld returned 1 exit status```.

**Reply by GiudGiud** [reply]
Can you grep for XXXTestApp in the head of the directory?
We have to find what s referring to this app in your files. This really looks like a test object.
